ecc_enclave/enclave/shim.cpp

```diff
@@ -51,8 +51,26 @@ void get_state(
         *val_len = 0;
         return;
     }
+    if (encoded_cipher_len > sizeof(encoded_cipher))
```

Comment:
Hmm, this shouldn't really happen as we pass to get_public_state sizeof(encoded_cipher) as maximum buffer length. So if that happens, there is a bug elsewhere and we should fix it there? Looking at the code, though, not clear how this could happen ...
Maybe the issue is that we never initialized encoded_cipher_len and due to some errors it never gets assigned? Maybe we should set it to zero in get_*_state before doing the ocall?

Reply:
Right, it should not happen.
However, as `encoded_cipher_len` is input data, it seems appropriate to perform input validation before building the string.

Reply:
input validation is done down in the SDK. Don't see any reason to do it again here and clutter the code. And if so, we should do it everywhere else as well?

Reply:
the input **to this function** is not validated. I don't see why you would not want to do it. Actually, we shall check the buffers as well.

Reply:
i'm missing something here: neither encoded_cipher_len nor encoded_cipher are inputs but local variables, so you are only checking return values from our own get_public_state function?

Reply:
right -- though I don't care that it is our own `get_public_state`.
Also, the `get_public_state` grabs this value from the ocall and, again, does not validate the range of the value.

Reply:
but then we should move it to get_public_state, not get_state?

Reply:
I think that we should rather implement the check there too. Yes, it is redundant, but I don't know how these functions will evolve over time. So I prefer to follow "per function" due diligence.

Reply:
after thinking again what the SDK does and not does, i think adding the check immediately after the ocalls, e.g., in get_public_state (and also after the ocall to ocall_get_state_by_partial_composite_key) is actually a good idea as the SDK almost certainly does not check this (i was wrong on this in my earlier comment). However, i would object to do it get_state where we call our own function as this is redundant and blurs the really important place for the checks, i.e., immediately after the ocall returns,

Reply:
(this is related to #81 )